### PR TITLE
original datapoint year update

### PR DESCRIPTION
**Pull request by rfontanarosa**
part of #1126

### CODE REVIEW

File: client/pages/OriginalDataPoint/OriginalDataPoint.tsx [hunk 1]
```diff
@@ -28,13 +29,15 @@ const OriginalDataPoint = () => {
         cycleName,
       })
     )
-    return () => dispatch(OriginalDataPointActions.reset())
+    dispatch(OriginalDataPointActions.reset())
```
Comment (sorja) [inline client/pages/OriginalDataPoint/OriginalDataPoint.tsx:32]: why you removed unmount action

File: client/pages/OriginalDataPoint/OriginalDataPoint.tsx [hunk 2]
```diff
@@ -28,13 +29,15 @@ const OriginalDataPoint = () => {
...
   }, [])
 
   if (!originalDataPoint) {
     return null
   }
 
+  if (originalDataPoint.countryIso !== countryIso) return null
```
Comment (sorja) [inline client/pages/OriginalDataPoint/OriginalDataPoint.tsx:39]: We need to discuss this url mismatch in daily on tuesday (reminder)

File: client/pages/OriginalDataPoint/OriginalDataPoint.tsx (hunk 1 above)
Comment (rfontanarosa) [reply client/pages/OriginalDataPoint/OriginalDataPoint.tsx]: my bad. there was a return type error and I let visual studio fix the issue. I've just pushed the fix.

File: client/pages/OriginalDataPoint/OriginalDataPoint.tsx (hunk 2 above)
Comment (rfontanarosa) [reply client/pages/OriginalDataPoint/OriginalDataPoint.tsx:41]: ok

File: client/pages/OriginalDataPoint/OriginalDataPoint.tsx [hunk 3]
```diff
@@ -28,13 +29,17 @@ const OriginalDataPoint = () => {
...
+    return () => {
+      dispatch(OriginalDataPointActions.reset())
+    }
   }, [])
 
   if (!originalDataPoint) {
     return null
   }
 
+  if (originalDataPoint.countryIso !== countryIso) return null
```
Comment (minotogna) [inline client/pages/OriginalDataPoint/OriginalDataPoint.tsx:41]: Remove this check 
We'll need to use only id in the url

Comment (minotogna) [inline client/pages/OriginalDataPoint/OriginalDataPoint.tsx:41]: add a redirect to country home page in this case

**Review by minotogna — CHANGES_REQUESTED** [review]

File: client/pages/OriginalDataPoint/OriginalDataPoint.tsx [hunk 4]
```diff
@@ -34,22 +32,22 @@ const OriginalDataPoint: React.FC = () => {
...
         <h1 className="title align-left">{i18n.t('nationalDataPoint.nationalDataPoint')}</h1>
         <ButtonBar canEditData={canEditData} />
       </div>
 
-      <YearSelection canEditData={canEditData} />
-      <DataSources canEditData={canEditData} />
-      <NationalClasses canEditData={canEditData} />
-      <OriginalData canEditData={canEditData} />
-      <Comments canEditData={canEditData} />
+      {originalDataPoint && (
```
Comment (sorja) [inline client/pages/OriginalDataPoint/OriginalDataPoint.tsx:42]: if !odp return null

Comment (rfontanarosa) [reply client/pages/OriginalDataPoint/OriginalDataPoint.tsx:42]: it doesn't work

Comment (minotogna) [reply client/pages/OriginalDataPoint/OriginalDataPoint.tsx:42]: make it work then 😏

File: client/pages/OriginalDataPoint/OriginalDataPoint.tsx [hunk 5]
```diff
@@ -28,28 +27,30 @@ const OriginalDataPoint = () => {
...
+    return () => {
+      dispatch(OriginalDataPointActions.reset())
+    }
   }, [])
 
-  if (!originalDataPoint) {
-    return null
-  }
-
   return (
```
Comment (sorja) [inline client/pages/OriginalDataPoint/OriginalDataPoint.tsx:35]: Above this:

`if (!originalDataPoint) return null`

File: server/repository/originalDataPoint/getReservedYears.ts [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+
+  const schemaName = Schemas.getNameCycle(assessment, assessmentCycle)
+  const years = await client.many(
+    `
+        select year from ${schemaName}.original_data_point where country_iso = $1
+    `,
+    [countryIso]
+  )
+
+  return years.map(({ year }) => year)
```
Comment (minotogna) [inline server/repository/originalDataPoint/getReservedYears.ts:23]: you could use client .map

Comment (minotogna) [reply server/repository/originalDataPoint/getReservedYears.ts:23]:
```typescript
return client.map<number>(
     `
         select year from ${schemaName}.original_data_point where country_iso = $1
     `,
     [countryIso],
     ({ year }) => year
   )
```

File: server/controller/assessment/getReservedYears.ts [hunk 7]
```diff
@@ -0,0 +1,15 @@
+import { BaseProtocol, DB } from '@server/db'
+import { AssessmentRepository, OriginalDataPointRepository } from '@server/repository'
+import { CountryIso } from '@meta/area'
+import { AssessmentName } from '@meta/assessment'
+
+export const getReservedYears = async (
+  props: { assessmentName: AssessmentName; cycleName: string; countryIso: CountryIso },
```
Comment (minotogna) [inline server/controller/assessment/getReservedYears.ts:7]: pass the assessment and cycle here, not primitive props

File: client/pages/OriginalDataPoint/OriginalDataPoint.tsx [hunk 8]
```diff
@@ -1,22 +1,26 @@
...
-const OriginalDataPoint = () => {
+const OriginalDataPoint: React.FC = () => {
   const { i18n } = useTranslation()
   const dispatch = useAppDispatch()
+  const history = useHistory()
+
   const originalDataPoint = useOriginalDataPoint()
-  const { assessmentName, cycleName, odpId } = useParams<{ assessmentName: string; cycleName: string; odpId: string }>()
+  const { assessmentName, cycleName, odpId, countryIso } =
+    useParams<{ assessmentName: string; countryIso: string; cycleName: string; odpId: string }>()
```
Comment (minotogna) [inline client/pages/OriginalDataPoint/OriginalDataPoint.tsx:22]: use the hooks

**Review by minotogna — CHANGES_REQUESTED** [review]

File: server/api/assessment/updateOriginalDataPoint.ts [hunk 9]
```diff
@@ -0,0 +1,25 @@
...
+import { AssessmentController } from '@server/controller'
+
+export const updateOriginalDataPoint = async (req: Request, res: Response) => {
+  try {
+    const { name, cycleName } = req.params
+    const { originalDataPoint } = req.body
+
+    const assessment = await AssessmentController.getOne({ name })
+
+    const assessmentCycle = assessment.cycles.find((cycle) => cycle.name === cycleName)
```
Comment (minotogna) [inline server/api/assessment/updateOriginalDataPoint.ts:12]: use AssessmentController.getOneWithCycle

**Review by minotogna — APPROVED** [review]

**Pull request merged**